packages/yarnpkg-pnpify/sources/generateSdk.ts

```diff
@@ -147,168 +262,91 @@ class Wrapper {
-
-      function addZipPrefix(str) {
-        if (isAbsolute(str) && str.match(/\\.zip\\//) && !str.match(/^zip:/)) {
-          return \`zip:\${str}\`;
-        } else {
-          return str;
-        }
-      }
+  const userEditors = editors ?? new Set<SupportedEditor>();
+  validateEditors(userEditors);
 
-      function removeZipPrefix(str) {
-        return str.replace(/^zip:/, \`\`);
-      }
-    };
-  `;
+  const editorsFile = new EditorsFile();
+  if (hasEditorsFile)
+    await editorsFile.loadFile(editorsFilePath);
+  const pnpifiedEditors = editorsFile.editors;
 
-  const wrapper = new Wrapper(`typescript` as PortablePath, {pnpApi, target});
+  const allEditors = new Set([
+    ...userEditors,
+    ...pnpifiedEditors,
+  ]);
 
-  await wrapper.writeManifest();
+  if (base)
+    report.reportInfo(null, `Installing the base SDK...`);
 
-  await wrapper.writeBinary(`bin/tsc` as PortablePath);
-  await wrapper.writeBinary(`bin/tsserver` as PortablePath);
+  if (allEditors.size === 0 && !hasEditorsFile && !base)
+    throw new UsageError(`No editors have been provided as arguments and no existing editors could be found inside the ${chalk.magenta(EditorsFile.fileName)} file. Make sure to use \`yarn pnpify --sdk <editors>\` or \`yarn pnpify --sdk base\` inside non-pnpified projects`);
 
-  await wrapper.writeFile(`lib/tsc.js` as PortablePath);
-  await wrapper.writeFile(`lib/tsserver.js` as PortablePath, {wrapModule: tsServerMonkeyPatch});
-  await wrapper.writeFile(`lib/typescript.js` as PortablePath);
+  if (xfs.existsSync(targetFolder)) {
+    report.reportInfo(null, `Cleaning up the existing SDK files...`);
+    await xfs.removePromise(targetFolder);
+  }
 
-  await addVSCodeWorkspaceSettings(pnpApi, {
-    [`typescript.tsdk`]: npath.fromPortablePath(
-      ppath.dirname(
-        wrapper.getProjectPathTo(
-          `lib/tsserver.js` as PortablePath,
-        ),
-      ),
-    ),
-    [`typescript.enablePromptUseWorkspaceTsdk`]: true,
-  });
-};
+  editorsFile.editors = allEditors;
+  await editorsFile.persist(targetFolder);
 
-export const generateStylelintWrapper = async (pnpApi: PnpApi, target: PortablePath) => {
-  const wrapper = new Wrapper(`stylelint` as PortablePath, {pnpApi, target});
+  report.reportInfo(null, `Installing fresh SDKs for ${chalk.magenta(projectRoot)}:`);
+  report.reportSeparator();
 
-  await wrapper.writeManifest();
+  if (allEditors.size > 0) {
+    report.reportInfo(null, `Editors:`);
+    for (const editor of userEditors)
+      report.reportInfo(MessageName.UNNAMED, `${chalk.green(`✓`)} ${getDisplayName(editor)} (user)`);
+    for (const editor of pnpifiedEditors)
+      if (!userEditors.has(editor))
+        report.reportInfo(MessageName.UNNAMED, `${chalk.green(`✓`)} ${getDisplayName(editor)} (editors file)`);
+    report.reportSeparator();
+  }
 
-  await wrapper.writeBinary(`bin/stylelint.js` as PortablePath);
-  await wrapper.writeFile(`lib/index.js` as PortablePath);
+  const EDITOR_SDKS = [
```

Comment:
No caps since it's not a constant (I mean, it's a local one).

Reply:
Also, I don't understand what this variable is for - I guess it'll be clearer with more integrations, but it's at least worth a comment or two.